src/Pluggable.groovy

```diff
@@ -0,0 +1,31 @@
+trait Pluggable<T> {
+    private static plugins = []
+    private appliedPlugins = []
+
+    public applyPlugins() {
+        def remainingPlugins = plugins - appliedPlugins
+
+        for (T plugin in remainingPlugins) {
+            plugin.apply(this)
+            appliedPlugins << plugin
+        }
+    }
+
+    public static void addPlugin(T plugin) {
+        plugins << plugin
+    }
+
+    public static void setPlugins(plugins) {
+        this.plugins = plugins
+    }
+
+    public static getPlugins() {
+        return plugins
+    }
+
+    public String toString() {
+        applyPlugins()
+        // Chain the toString call to the next trait in the chain
+        return super.toString()
+    }
```

Comment:
Chaining toString() seems fraught with peril.

Conceptual speaking, toString() has nothing to do with the idea of being Pluggable.  toString is totally incidental to the way that it's being used in TerraformCommands.  In the future, I'd like to be able to extend Pluggable to other areas of terraform-pipeline, and I think toString() will hinder that future work. 

As a first-cut, I think toString() should be left out of the interface.

Reply:
I guess I'll rephrase to say, I'd explicitly NOT like to include toString() to be part of contact.  And this dove tails with my comment above, about wanting to extend the idea of a `Pluggable` to Stages as well.  Applying plugins is coincidental to TerraformCommands, and would be inappropriate for a Stage.  Hence, I'd prefer to leave toString out of the contract of Pluggables to *enable* future application to Stages.

Reply:
toString() is reverted to it's previous behavior, and not part of TerraformCommand or Pluggable